lib/Result.ts

```diff
@@ -418,6 +419,25 @@ export const promiseResult = <Success, Failure>(
   })
 }
 
+/**
+ * Extracts the success value of a given result. Ex.
+ * 
+ * ```ts
+ * const getUser = (user: {name: string, id: number}) =>
+ *  success(user)
+ *    .mapSuccess(({ name, id }) => ({ name, id, newField: `${name}${id}` }))
+ * 
+ * type transformedUserType = ExtractSuccess<ReturnType<typeof getUser>>
+ * //transformedUserType is derived as {name: string, id: number, newField: string}
+ * ```
+ */
+export type ExtractSuccess<T> = T extends AwaitableResult<infer U, any> ? U : never;
+
+/**
+ * Extracts the failure value of a given result. See {@link ExtractSuccess}
+ */
+export type ExtractFailure<T> = T extends AwaitableResult<infer U, any> ? U : never;
```

Comment:
```suggestion
export type ExtractFailure<T> = T extends AwaitableResult<any, infer U> ? U : never;
```
The 2 types are exactly the same, this should fix that.